gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_ui.ipynb

```diff
@@ -0,0 +1,907 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "rJG1pVZO317x"
+      },
+      "outputs": [],
+      "source": [
+        "INSTRUCTION_TEMPLATE = \"Answer the following question. Let's think step by step.\\n\"  # @param {type:\"string\"}\n",
+        "CONTEXT_TASK_TEMPLATE = (\n",
+        "    \"Question: {{question}}\\n\\nAnswer:{{target}}\"  # @param {type:\"string\"}\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5y-cmg0TQP6v"
+      },
+      "source": [
+        "# Step 2: Input your data\n",
+        "To optimize the model, provide a CSV or JSONL file containing labeled validation samples\n",
+        "* Focus on examples that specifically demonstrate the issues you want to address.\n",
+        "* Recommendation: Use 50-100 distinct samples for reliable results. However, the tool can still be effective with as few as 5 samples.\n",
+        "\n",
+        "For prompt translation:\n",
+        "* Consider using the source model to label examples that the target model struggles with, helping to identify areas for improvement.\n",
+        "\n",
+        "Learn more about setting up your CSV or JSONL file as input [TODO: add link]."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "mfgi_oR6tTIB"
+      },
+      "outputs": [],
+      "source": [
+        "# @markdown **Project setup**: <br/>\n",
+        "PROJECT_ID = \"vertex-research-solutions\"  # @param {type:\"string\"}\n",
+        "LOCATION = \"us-central1\"  # @param {type:\"string\"}\n",
+        "# @markdown * GCS path of your bucket, e.g., gs://prompt_translation_demo, used to store all artifacts.\n",
+        "BUCKET = \"gs://prompt_design_demo\"  # @param {type:\"string\"}\n",
+        "# @markdown * Specify the input data path inside your bucket.\n",
+        "INPUT_DATA_PATH = \"movie_recommendation.jsonl\"  # @param {type:\"string\"}"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ucebZHkHRxKH"
+      },
+      "source": [
+        "# Step 3: Configure optimization settings\n",
+        "The optimization configs are defaulted to the values that are most commonly used and which we recommend using initially. Refer here [TODO:add link] to learn more about the different parameters settings and how to best utilize them."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "B2R3P8mMvK9q"
+      },
+      "outputs": [],
+      "source": [
+        "TARGET_MODEL = \"gemini-1.5-flash-001\"  # @param [\"gemini-1.0-pro-001\", \"gemini-1.0-pro-002\", \"gemini-1.5-flash-001\", \"gemini-1.5-pro-001\", \"gemini-1.0-ultra-001\", \"text-bison@001\", \"text-bison@002\", \"text-bison32k@002\", \"text-unicorn@001\"]\n",
```

Comment:
I like this dropdown menu! Although I would recommend against including the PaLM models as target models.